src/swsssdk/configdb.py

```diff
@@ -49,19 +47,26 @@ def __init__(self, decode_responses=True, **kwargs):
            'namespace' is implicitly passed to the parent SonicV2Connector class.
         """
         super(ConfigDBConnector, self).__init__(**kwargs)
+        # Trick: to achieve static/instance method "overload", we must use initize the function in ctor
+        # ref: https://stackoverflow.com/a/28766809/2514803
+        self.serialize_key = self._serialize_key
```

Comment:
What are we trying to solve here ?
Per link pasted above (Thanks for reference), all we need to declare instance variation for the static methods as suggested -- more a hack. But this code looks not  the same. Can you please help understand?

I guess, (based on  those static methods) the root cause could be Class & instance variable values for *_SEPARATOR could be different. 
Fix: 
In db_connect, you may add the following lines and see if that fixes the issue.

ConfigDBConnector.TABLE_NAME_SEPARATOR = self.TABLE_NAME_SEPARATOR
ConfigDBConnector.KEY_SEPARATOR = self.KEY_SEPARATOR

Reply:
Static method `serialize_key` does not make sense because it does not take the instance separator. So I implement the instance method `serialize_key`.
However the statice method is used even before `do_connnect` was introduced. They are assuming separator `|`. So I must keep both static method and instance method to make all applications happy.
It is not straightforward to `overload` this way in python. I must use this trick.


---
In reply to: [549801673](https://github.com/Azure/sonic-py-swsssdk/pull/99#discussion_r549801673) [](ancestors = 549801673)

Reply:
My only concern: Code duplication. Two serialize & de-serialize methods. If the code can be combined, with separator passed in, it will be  better. Or some other more smarter pythonic way.

e.g. 
@static 
serialize(key, sep="|")

def _serialize(self, key)
  ConfigDBConnector,serialize(key, self.KEY_SEPARATOR)